addons/sale/models/sale_order.py

```diff
@@ -768,8 +769,11 @@ def action_confirm(self):
                 'It is not allowed to confirm an order in the following states: %s'
             ) % (', '.join(self._get_forbidden_state_confirm())))
 
+        mapped_subscriptions = defaultdict(lambda: self.env['sale.order'])
         for order in self.filtered(lambda order: order.partner_id not in order.message_partner_ids):
-            order.message_subscribe([order.partner_id.id])
+            mapped_subscriptions[order.partner_id.id] += order
```

Comment:
We may as well remove the `for.. in x.filtered` here:
```suggestion
            if order.partner_id in order.message_partner_ids:
                continue
            mapped_subscriptions[order.partner_id.id] += order
```